tests/cli.rs

```diff
@@ -3817,3 +3817,40 @@ fn reason_phrase_is_preserved() {
 
         "#});
 }
+
+#[test]
+fn error_code_416_is_ignored_when_resuming_download() {
+    let server = server::http(|_req| async move {
+        hyper::Response::builder()
+            .status(416)
+            .header(hyper::header::CONTENT_TYPE, "image/png")
+            .body("".into())
+            .unwrap()
+    });
+    
+    get_command()
+        .arg(server.base_url())
+        .arg("--download")
+        .arg("--continue")
+        .args(["--output", "/dev/null"])
+        .assert()
+        .success();
+}
+
+#[test]
+fn error_code_416_is_not_ignored_when_not_resuming_download() {
+    let server = server::http(|_req| async move {
+        hyper::Response::builder()
+            .status(416)
+            .header(hyper::header::CONTENT_TYPE, "image/png")
+            .body("".into())
+            .unwrap()
+    });
+    
+    get_command()
+        .arg(server.base_url())
+        .arg("--download")
+        .args(["--output", "/dev/null"])
+        .assert()
+        .failure();
```

Comment:
Can you also check the exit status (`.code(4)`) and check that stderr is empty to make sure that it didn't fail in some unexpected other way?